amplfi/train/data/waveforms/generator/cbc.py

```diff
@@ -1,153 +1,41 @@
-import math
-from functools import partial
-from typing import Any, Callable, Dict
+from typing import Callable
 
 import torch
+from ml4gw.waveforms.generator import TimeDomainCBCWaveformGenerator
 
 from .generator import WaveformGenerator
 
 
-class FrequencyDomainCBCGenerator(WaveformGenerator):
-    """
-    A torch module for generating waveforms on the fly.
-
-    Args:
-        waveform:
-            A callable that returns cross and plus polarizations
-            given a set of parameters.
-        duration:
-            The duration of the waveform in seconds
-        sample_rate:
-            Sample rate of the waveform in Hz
-        f_min:
-            The minimum frequency of the waveform in Hz
-        f_max:
-            The maximum frequency of the waveform in Hz
-        ringdown_duration:
-            The duration of time in seconds to roll the fft'd
-            waveform to the left to join the coalescence and ringdown.
-            This will place the coalescence time `ringdown_duration` seconds
-            from the right edge of the waveform. Defaults to 0.5.
-        waveform_arguments:
-            A dictionary of fixed arguments to pass to the waveform model,
-            e.g. `f_ref` for CBC waveforms
-    """
-
+class CBCGenerator(WaveformGenerator):
     def __init__(
         self,
         *args,
         approximant: Callable,
```

Comment:
Let's add a good docstring to this, giving an example of the callable (possibly `ml4gw.waveforms.IMRPhenomD()` and similar?). Does the waveform instance be in the correct device at the time of calling the init?

Reply:
Docstring added. The whole `amplfi.train.waveforms.generator.WaveformGenerator` is a torch module that should get sent to the proper device, and thus any sub-torch modules like the waveform instance should get sent there too.